backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Integration/Clients/ConnectionClient.cs

```diff
@@ -50,20 +52,25 @@ public ConnectionClient(
         }
 
         /// <inheritdoc />
-        public async Task<HttpResponseMessage> PostNewRightHolderConnection(Guid party, Guid to, CancellationToken cancellationToken = default)
+        public async Task<Guid> PostNewRightHolderConnection(Guid party, Guid? to, PersonInput personInput = null, CancellationToken cancellationToken = default)
         {
-            string endpointUrl = $"enduser/connections?party={party}&from={party}&to={to}";
+            string endpointUrl = $"enduser/connections?party={party}&from={party}" + (to != null ? $"&to={to}" : string.Empty);
             string token = JwtTokenUtil.GetTokenFromContext(_httpContextAccessor.HttpContext, _platformSettings.JwtCookieName);
 
-            var httpResponse = await _client.PostAsync(token, endpointUrl, null);
+            StringContent requestBody = personInput != null ? new StringContent(JsonSerializer.Serialize(personInput, _serializerOptions), Encoding.UTF8, "application/json") : null;
+
+            var httpResponse = await _client.PostAsync(token, endpointUrl, requestBody);
+
+            var content = await httpResponse.Content.ReadAsStringAsync();
 
             if (!httpResponse.IsSuccessStatusCode)
             {
                 _logger.LogError($"Unexpected http response. Status code: {httpResponse.StatusCode}, Reason: {httpResponse.ReasonPhrase}");
                 throw new HttpStatusException("Unexpected http response.", "Unexpected http response.", httpResponse.StatusCode, null, httpResponse.ReasonPhrase);
             }
 
-            return httpResponse;
+            AddConnectionResponse response = JsonSerializer.Deserialize<AddConnectionResponse>(content, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+            return response.Id;
```

Comment:
Trying to add a new person, i'm passed to a new page, but i don't get a valid connection.

It looks like "ConnectionId" rather than the "PersonId" that get s returned? Should this instead be:  

```suggestion
            return response.ToId;
```

Reply:
Oops! 🙈
I thought the reason why I didn't get a valid page was because the connectionAPI was unstable, but you're right - it absolutely should be the `ToId` and not the `Id` that should be used here!
Thank you for noticing! 🙌